docs/applications/consuming-android-libraries.md

```diff
@@ -0,0 +1,196 @@
+components into a single optimized `.so` library file.  The managed code can be easily configured to
+use such a megazord without significant changes.
+
+There are two tasks we want to arrange.  First, we want to substitute component modules for the single
+megazord module; second, we want to arrange for native Rust code to be available to JVM unit tests.
+
+Both tasks are handled by the [megazord-gradle](https://github.com/mozilla/megazord-gradle) Gradle
+plugin; see that page for details on how to apply the plugin to your build configuration.
+
+## Consuming megazords
+
+To consume a specific megazord module, use something like:
+
+```groovy
+appservices {
+    defaultConfig {
+        // Megazord in all variants.  The default is to not megazord.
+        megazord = 'org.mozilla.appservices.megazord'
+        // With no configuration, the default is to enable unit tests for all variants.
+        enableUnitTests = true
+    }
+
+    // or:
+
+    buildTypes {
+        debug {
+            megazord = 'org.mozilla.appservices.megazord'
+            enableUnitTests = true
+        }
+        // Do not megazord or enable unit tests in release build type.
+    }
+
+    // or:
+
+    productFlavors {
+        stage {
+            megazord = 'org.mozilla.appservices.megazord'
+            enableUnitTests = true
+        }
+        // Do not megazord or enable unit tests in other product flavors.
+    }
+
+    // or:
+
+    variants {
+        stageDebug {
+            megazord = 'org.mozilla.appservices.megazord'
+            enableUnitTests = true
+        }
+        // Do not megazord or enable unit tests in other variants.
+    }
+}
+```
+
+### Configuring the consuming application
+
+The megazord modules expose a single additional JVM class, like
+`org.mozilla.appservices.composites.{Lockbox,ReferenceBrowser}Composition`.  That class has a single
+static `init()` method that consuming applications should invoke in their `Application.onCreate`,
+like:
+
+```xml
+<manifest>
+    <application android:name=".Application" ...>
+    </application>
+    ...
+</manifest>
+```
+
+and:
+
+```java
+public class Application extends android.app.Application {
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        LockboxComposition.init();
+    }
+
+    ...
+}
+```
+
+This `init()` method sets some Java system properties that tell the component modules what native
+code library contains the underlying component native code.
+
+### Pseudo-code details
+
+For each Android variant `variant`, the corresponding Gradle configurations
+(`variant.{compileConfiguration,runtimeConfiguration}`) have module substitutions applied, like:
+
+```groovy
+// Pseudo-code!
+configuration.resolutionStrategy.dependencySubstitution.all { dependency ->
+    if (dependency.isComponentModule()) {
+        dependency.useTarget('org.mozilla.appservices.megazord:megazord:...')
+    }
+}
+```
```

Comment:
I read this as a kind of explanatory example, not something I actually have to do in my code; is that accurate?

Reply:
If it *is* something I need to do in my consuming application, perhaps it's worth a checklist at the top of the doc with links to the details sections, something like:

```
In order to consume a megazord build you need to:
* Figure out which one to use.
* Add blah-blah to your gradle config.
* Call `init()` in `onCreate()`
* ...etc...
```

Reply:
So: I'm putting the Gradle plugin directly into the a-s repo.  The documentation will be split between here (for consumers) and in the tree (for developers), which I think will clarify.  In addition, I like the checklist idea -- should have done that to start, honestly -- and I'll build it in.

Thanks!